app/scripts/metamask-controller.js

```diff
@@ -2779,14 +2787,15 @@ export default class MetamaskController extends EventEmitter {
   }
 
   /**
-   * A method for creating a provider that is safely restricted for the requesting domain.
+   * A method for creating a provider that is safely restricted for the requesting subject.
+   *
    * @param {Object} options - Provider engine options
    * @param {string} options.origin - The origin of the sender
    * @param {string} options.location - The full URL of the sender
+   * @param {string} options.subjectType - The type of the sender subject.
```

Comment:
Nit: It's not really the type of the _object_. It's the type of the sender.

There is a sender object (e.g. in `setupProviderConnection`) but that doesn't differ between subject types.

```suggestion
   * @param {string} options.subjectType - The type of the sender.
```

Reply:
It's sender _subject_, not object!

Reply:
Welp. Not sure how I misread that, even during editing.